vision/Object Detection/standard_obj_text.py

```diff
@@ -2,76 +2,116 @@
+    def __init__(self) -> None:
+        # related to text detection
+        self.rotated_img: npt.NDArray[np.uint8] = np.array([])
+        self.preprocessed: npt.NDArray[np.uint8] = np.array([])
 
-    def get_text_characteristics(self, img: np.ndarray, bounds: np.ndarray) -> tuple:
+        # related to text color
+        self.text_cropped_img: npt.NDArray[np.uint8] = np.array([])
+        self.kmeans_img: npt.NDArray[np.uint8] = np.array([])
+        self.color: Optional[str] = None
+
+        # related to text orientation
+        self.orientation: Optional[str] = None
+
+    def get_text_characteristics(
+        self, img: npt.NDArray[np.uint8], bounds: List[List[int]]
+    ) -> Tuple[Optional[str], Optional[str], Optional[str]]:
         """
         Gets the characteristics of the text on the standard object.
 
         Parameters
         ----------
-        img : np.ndarray
+        img : np.NDArray[np.uint8]
             image to find characteristics of text within
-        bounds : np.ndarray
+        bounds : List[List[int, int]]
             bounds of standard object containing text on in image
 
         Returns
         -------
-        tuple
-            characteristics of the text in the form (character, orientation, color)
+        (character, orientation, color) : Tuple[str, str, str]
+            characteristics of the text in the form
         """
         ## Get the character ##
-        characters = self._detect_text(img, bounds)
+        characters: List[Tuple[str, List[Tuple[int, int]]]] = self._detect_text(
+            img, bounds
+        )
         if len(characters) != 1:
             return (None, None, None)
+        character: str
+        char_bounds: List[Tuple[int, int]]
         character, char_bounds = characters[0]
 
         ## Get the orientation ##
-        orientation = self._get_orientation(img, char_bounds)
+        orientation: Optional[str] = self._get_orientation()
 
         ## Get the color of the text ##
-        color = self._get_text_color(img, char_bounds)
+        color: Optional[str] = self._get_text_color(char_bounds)
 
         return (character, orientation, color)
 
-    def _detect_text(self, img: np.ndarray, bounds: np.ndarray) -> list:
+    def _detect_text(
+        self, img: npt.NDArray[np.uint8], bounds: List[List[int]]
+    ) -> List[Tuple[str, List[Tuple[int, int]]]]:
         """
         Detect text within an image.
         Will return string for parameter odlc.alphanumeric
 
         Parameters
         ----------
-        img : np.ndarray
+        img : npt.NDArray[np.uint8]
             image to detect text within
-        bounds : np.ndarray
+        bounds : List[List[int, int]]
             array of tuple bounds (4 x-y coordinates)
 
         Returns
         -------
-        list
-            list containing detected characters and their bounds, Format of characters is ([bounds], 'character').
+        found_characters : List[Tuple[str, List[Tuple[int, int]]]]
```

Comment:
You might want to extrapolate this structural subtyping at least one level further. 
If you want, as a way to clean things up, you can utilize `TypeAlias` or `NewType` for things like `List[Tuple[int, int]]` since it is used quite a few times in this module, but you don't have to do this.
The goal here is to make it clear how a collection of collections of collections actually maps to the logical output format. (i.e. where is a found character and bounds in this complex return type, and what do bounds mean/look like.)